business/checkers/destinationrules/no_dest_checker.go

```diff
@@ -48,10 +50,16 @@ func (n NoDestinationChecker) Check() ([]*models.IstioCheck, bool) {
 										}
 									}
 									if !n.hasMatchingWorkload(fqdn.Service, stringLabels) {
-										validation := models.Build("destinationrules.nodest.subsetlabels",
-											"spec/subsets["+strconv.Itoa(i)+"]")
-										validations = append(validations, &validation)
-										valid = false
+										if n.isSubsetReferenced(dHost, innerSubset["name"].(string)) {
+											validation := models.Build("destinationrules.nodest.subsetlabels",
+												"spec/subsets["+strconv.Itoa(i)+"]")
+											validations = append(validations, &validation)
+											valid = false
+										} else {
+											validation := models.Build("destinationrules.nodest.subsetlabelsnoterferenced",
+												"spec/subsets["+strconv.Itoa(i)+"]")
+											validations = append(validations, &validation)
+										}
```

Comment:
What happens when the subset is not referenced but the subset has some matching workloads?
I guess that with this implementation the validation won't come up, right?

Reply:
This can be reflected in a test case as well. wdyt?

Reply:
+1 to cover by test the scenario when subset is not referenced but have matching workload, for regression purposes